### PR TITLE
Add powershell version telemetry

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

Leveraging the [vscode-extension-telemetry](https://github.com/Microsoft/vscode-extension-telemetry) extension, this adds a single additional data point... "PowerShell version". This helps us answer the question:

"Who is using VSCode? 
Is it Windows PowerShell, PowerShell Core?
Is anyone on v3/v4 using the stable version right now?"

This means we can better focus our efforts in developing and maintaining the PowerShell extension.

This node module honors the vscode mechanism for opting out of telemetry so we do not need to add any custom logic for that scenario.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [X] PR has a meaningful title
- [X] Summarized changes
- [ ] PR has tests
- [ ] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -35,24 +36,39 @@ import Settings = require("./settings");
...
     { language: "powershell", scheme: "untitled" },
 ];
 
 export function activate(context: vscode.ExtensionContext): void {
+    // create telemetry reporter on extension activation
+    telemetryReporter = new TelemetryReporter(PackageJSON.name, PackageJSON.version, AI_KEY);
 
-    const version = getCurrentVersion(context);
+    // If both extensions are enabled, this will cause unexpected behavior since both register the same commands
+    if (PackageJSON.name.toLowerCase() === "powerShell-preview"
```
Comment (rkeithhill) [inline src/main.ts]: You have a capital `S` in powerShell that will cause the toLowerCase comparison to always fail.

File: src/session.ts [hunk 2]
```diff
@@ -108,6 +112,11 @@ export class SessionManager implements Middleware {
         this.createStatusBarItem();
 
         this.powerShellExePath = this.getPowerShellExePath();
+        cp.exec(`${this.powerShellExePath} -v`, (error: cp.ExecException, stdout: string, stderr: string) => {
```
Comment (rkeithhill) [inline src/session.ts]: So Windows PowerShell won't work like you want.  :-(
```
04-18 18:58:30 64> powershell -v
Missing argument for parameter version.
```

Comment (TylerLeonhardt) [reply src/session.ts]: Gah... Does `-version` work?

Comment (rkeithhill) [reply src/session.ts]: There is the `this.versionDetails` field in session.ts.  You'd just have to wait until after it is set.  That is what we use to determine the version to display in the status bar.

Comment (rjmholt) [reply src/session.ts]: `-Version` in Windows PowerShell is to do this:

```
> powershell -version 2 -command '$PSVersionTable'

Name                           Value
----                           -----
CLRVersion                     2.0.50727.9136
BuildVersion                   6.1.7600.16385
PSVersion                      2.0
WSManStackVersion              2.0
PSCompatibleVersions           {1.0, 2.0}
SerializationVersion           1.1.0.1
PSRemotingProtocolVersion      2.1
```

(and values other than 2 are ignored)

Comment (rkeithhill) [reply src/session.ts]: That's worse:
```
04-18 18:59:17 66> powershell.exe --version
At line:1 char:3
+ --version
+   ~
Missing expression after unary operator '--'.
At line:1 char:3
+ --version
+   ~~~~~~~
Unexpected token 'version' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingExpressionAfterOperator
```
In v5.x, `-Version` expects you to specify a version to start.

Comment (rkeithhill) [reply src/session.ts]: You can always use:
```
powershell -encodedcommand JABQAFMAVgBlAHIAcwBpAG8AbgBUAGEAYgBsAGUALgBQAFMAVgBlAHIAcwBpAG8AbgAuAFQAbwBTAHQAcgBpAG4AZwAoACkA
```
:-)

Comment (rkeithhill) [reply src/session.ts]: I'd just use `this.versionDetails` that way we don't have to spin up a second PowerShell.

File: src/main.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/main.ts]: Lol... 🤦‍♂️

File: src/session.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/session.ts]: That's perfect

File: src/main.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/main.ts]:
```suggestion
	    if (PackageJSON.name.toLowerCase() === "powershell-preview"
```

**Comment by TylerLeonhardt** [comment]
Ok I've moved the location of the telemetry call to where @rkeithhill suggested. That's definitely the right place!

**Comment by TylerLeonhardt** [comment]
The one thing I'm not too sure about is how to stop debug builds from sending telemetry. Maybe it's just such a small number that it doesn't matter?

Thoughts?

**Comment by rjmholt** [comment]
I think it's probably too insignificant to care about, but otherwise the server can use `#if DEBUG` to generate those (it's a built-in definition I think).

The problem is getting that information in the client, which I think could only be achieved by either querying (or being sent at startup) information from the server, or if the client itself is built with a debug flag.

It's probably too much work to be worth it at this stage, but would be worth adding with greater investment later.

**Comment by TylerLeonhardt** [comment]
@rjmholt `#if DEBUG` in TypeScript?

**Comment by rkeithhill** [comment]
Isn't there a VSCode API to see if you're running in the extension dev host?

**Comment by rjmholt** [comment]
> @rjmholt #if DEBUG in TypeScript?

No, only server-side

**Comment by TylerLeonhardt** [comment]
@rkeithhill without doing something tricky like checking the file path, there doesn't seem to be a way to do that, unfortunately.

**Comment by rkeithhill** [comment]
Could you check the app title bar to see if it contains "Extension Development Host"?

**Comment by TylerLeonhardt** [comment]
Naturally David thought about this and a mechanism already existed :)

**Comment by rkeithhill** [comment]
That [VSCode issue](https://github.com/microsoft/vscode/issues/10272) is still open almost three years later.  :-(

**Pull request merged**